File: pyg-manager/pyg-manager-interface/src/main/java/com/pyg/manager/service/BrandService.java
package com.pyg.manager.service;

import com.pyg.pojo.TbBrand;
import com.pyg.utils.PageResult;
import com.pyg.utils.PygResult;

import java.util.List;

/**
 * @Title BrandService
 * @ProjectName pyg
 * @Description
 * @Author Hello.Ju
 * @Date 2019-07-15 23:04
 */
public interface BrandService {
    //查找所有的品牌
    List<TbBrand> findAll();

    //分页展示
    PageResult findPage(Integer page, Integer rows);

    //添加
    PygResult add(TbBrand tbBrand);

    //根据id查询品牌数据
    TbBrand findOne(Long id);

    PygResult update(TbBrand tbBrand);

    /**
     * 带条件查询
     * @param tbBrand
     * @param page
     * @param rows
     * @return
     */
    PageResult findBrandByPage(TbBrand tbBrand, Integer page, Integer rows);
}

File: pyg-manager-web/src/main/webapp/js/service/brandService.js
//品牌service
app.service("brandService", function ($http) {
    this.findAll = function () {
        return $http.get("../brand/findAll");
    };

    this.findPage = function (page, rows) {
        return $http.get('../brand/findPage?page=' + page + '&rows=' + rows);
    };

    this.findOne = function (id) {
        return $http.get("../brand/findOne?id="+id);
    };

    this.addOne = function (entity) {
        return $http.post("../brand/add", entity);
    };

    this.updateOne = function (entity) {
        return $http.post("../brand/update", entity);
    }

    this.dele = function (ids) {
        return $http.get("../brand/delete?ids="+ids)
    }

    this.search = function(page, rows, searchEntity) {
        return $http.post("../brand/search?page="+page+"&rows="+rows, searchEntity)
    }

});